compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -894,7 +898,8 @@ pub trait PrettyPrinter<'tcx>:
                     p!(print_def_path(did, args));
                     if !args.as_closure().is_valid() {
                         p!(" closure_args=(unavailable)");
-                        p!(write(" args={:?}", args));
+                        let v = args.iter().map(|arg| arg.to_string()).collect::<Vec<_>>();
```

Comment:
I wonder if we should have some adapter like `.print_as_list()` on `ty::GenericArgsRef` that does this for us.

Reply:
Good idea, I will add it.